ChatRPG/Services/GameInputHandler.cs

```diff
@@ -60,81 +66,118 @@ private void OnChatCompletionChunkReceived(bool isStreamingDone, string? chunk =
+                    if (campaign.Player.AdjustHealth(-opponentDmg))
+                    {
+                        combatMessageContent += "The player has died and their adventure ends.";
+                    }
+                    _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", opponent.Name, campaign.Player.Name, opponentDmg, campaign.Player.CurrentHealth, campaign.Player.MaxHealth);
                 }
-                _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", opponent.Name, campaign.Player.Name, opponentDmg, campaign.Player.CurrentHealth, campaign.Player.MaxHealth);
-            }
-            else
+                else
+                {
+                    combatMessageContent += $"The opponent will miss their next attack, dealing no damage. The player has {campaign.Player.CurrentHealth} health remaining.";
+                }
+
+                OpenAiGptMessage combatSystemMessage = new(ChatMessageRole.System, combatMessageContent);
+                conversation.Add(combatSystemMessage);
+                return _systemPrompts[systemPromptType];
+            default:
+                throw new ArgumentOutOfRangeException();
+        }
+    }
+
+    private async void DetermineAndPerformHurtOrHeal(Campaign campaign, ICollection<OpenAiGptMessage> conversation)
+    {
+        OpenAiGptMessage lastUserMessage = conversation.Last(m => m.Role.Equals(ChatMessageRole.User));
+        string hurtOrHealString = await _llmClient.GetChatCompletion(new List<OpenAiGptMessage>() { lastUserMessage }, _systemPrompts[SystemPromptType.HurtOrHeal]);
+        _logger.LogInformation("Hurt or heal response: {hurtOrHealString}", hurtOrHealString);
+        OpenAiGptMessage hurtOrHealMessage = new(ChatMessageRole.Assistant, hurtOrHealString);
+        LlmResponse? hurtOrHealResponse = hurtOrHealMessage.TryParseFromJson();
+        string hurtOrHealMessageContent = "";
+        Random rand = new Random();
+        if (hurtOrHealResponse?.Heal == true)
+        {
+            int healAmount = rand.Next(PlayerHealMin, PlayerHealMax);
+            campaign.Player.AdjustHealth(healAmount);
+            hurtOrHealMessageContent +=
+                $"The player heals {healAmount} health points, setting them at {campaign.Player.CurrentHealth} health points. Mention these numbers in your response.";
+        }
+
+        if (hurtOrHealResponse?.Hurt == true)
+        {
+            int dmgAmount = rand.Next(PlayerDmgMin, PlayerDmgMax);
+            hurtOrHealMessageContent += $"The player hurts themselves for {dmgAmount} damage. The player has {(campaign.Player.CurrentHealth - dmgAmount < 0 ? 0 : campaign.Player.CurrentHealth - dmgAmount)} health remaining. Mention these numbers in your response.";
```

Comment:
Her kan du også gøre det samme som i kommentaren jeg smed ind ovenfor, altså gem `campaign.Player.AdjustHealth(-dmgAmount)` i en boolean, og så append beskeden her bagefter. Så slipper du for de trælse conditionals.

Reply:
donerino